addOns/reveal/src/test/java/org/zaproxy/zap/extension/reveal/ExtensionRevealUnitTest.java

```diff
@@ -0,0 +1,159 @@
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import java.util.Locale;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.proxy.ProxyListener;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.utils.I18N;
+
+/** Unit test for {@link ExtensionReveal}. */
+class ExtensionRevealUnitTest {
+
+    private static final String DEFAULT_BODY =
+            "<html><head></head><body><H1>Some Heading</H1></body></html>";
+    private ExtensionReveal extension;
+
+    @BeforeEach
+    void setUp() {
+        Constant.messages = new I18N(Locale.ROOT);
+        extension = new ExtensionReveal();
+        extension.init();
+    }
+
+    @AfterEach
+    void cleanUp() {
+        Constant.messages = null;
+    }
+
+    @Test
+    void shouldAddProxyListenerAndApiImplementorOnHook() {
+        // Given
+        ExtensionHook extensionHook = mock(ExtensionHook.class);
+        // When
+        extension.hook(extensionHook);
+        // Then
+        ArgumentCaptor<ProxyListener> plArg = ArgumentCaptor.forClass(ProxyListener.class);
+        verify(extensionHook).addProxyListener(plArg.capture());
+        assertThat(plArg.getValue(), is(notNullValue()));
+
+        ArgumentCaptor<RevealAPI> apiArg = ArgumentCaptor.forClass(RevealAPI.class);
+        verify(extensionHook).addApiImplementor(apiArg.capture());
+        assertThat(apiArg.getValue(), is(notNullValue()));
+    }
+
+    @Test
+    void shouldBeUnloadable() {
+        // Given / When
+        boolean canUnload = extension.canUnload();
+        // Then
+        assertThat(canUnload, is(equalTo(true)));
+    }
+
+    @Test
+    void shouldHaveSameContentLengthOnUnmodifiedMessage() {
+        // Given
+        HttpMessage msg = createMessage();
+        // When
+        int contentLengthBefore =
+                Integer.parseInt(
+                        msg.getResponseHeader().getHeaderValues(HttpHeader.CONTENT_LENGTH).get(0));
+        extension.revealFields(msg);
+        int contentLengthAfter =
+                Integer.parseInt(
+                        msg.getResponseHeader().getHeaderValues(HttpHeader.CONTENT_LENGTH).get(0));
+        // Then
+        assertThat(contentLengthAfter, is(equalTo(contentLengthBefore)));
+        assertThat(contentLengthAfter, is(equalTo(msg.getResponseBody().length())));
```

Comment:
Why not hardcode the values?

Reply:
🤷‍♂️
1) I was lazy.
2) Seemed more maintenance friendly.

Will tweak.

Reply:
I'd not check the before, the content is not mutable for the need to check them both.